libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -458,3 +461,16 @@ void ContractLevelChecker::checkPayableFallbackWithoutReceive(ContractDefinition
 				SecondarySourceLocation{}.append("The payable fallback function is defined here.", fallback->location())
 			);
 }
+
+void ContractLevelChecker::checkStateVariablesOffsets(ContractDefinition const& _contract)
+{
+	try
+	{
+		// dry run, to make sure it succeeds later
+		ContractType(_contract).stateVariables();
+	}
```

Comment:
We could
* drag `ErrorReporter` down to the place where the error occurs;
* refactor to return an error (`Error`, `bool`, ...);
* throw (`Error`, custom exception).

None looks ideal, I would like to see more cases to develop a pattern. While in doubts, I've picked the shortest implementation.

Reply:
What is your problem in using the `ErrorReporter` interface? I'd just like to understand your point here. :-)

Reply:
> What is your problem in using the ErrorReporter interface?

Sometimes a function where an error occurs does not have access to the `ErrorReporter`, while the caller (might be few stack frames down) has.